src/sentry/middleware/superuser.py

```diff
@@ -1,8 +1,39 @@
 from __future__ import absolute_import
 
-from sentry.auth.utils import is_active_superuser
+from django.conf import settings
+
+from sentry.auth.superuser import is_active_superuser, logger, Superuser
 
 
 class SuperuserMiddleware(object):
     def process_request(self, request):
+        # This avoids touching user session, which means we avoid
+        # setting `Vary: Cookie` as a response header which will
+        # break HTTP caching entirely.
+        self.__skip_caching = request.path_info.startswith(settings.ANONYMOUS_STATIC_PREFIXES)
+        if self.__skip_caching:
+            return
+
+        su = Superuser(request)
+
         request.is_superuser = lambda: is_active_superuser(request)
```

Comment:
Afaik we have another one of these elsewhere for rest framework.

Reply:
grepped on this pattern and fixed a few other cases that were like this

tests still aren't passing, but thats likely because a bunch of our tests dont setup the superuser fixture as expected anymore. i'll be abstracting that so its easy, but likely make it opt-in (e.g. ``login_as(..., superuser=True)``)